src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -104,6 +105,18 @@ public sealed class EmitOptions : IEquatable<EmitOptions>
         /// </summary>
         public string? RuntimeMetadataVersion { get; private set; }
 
+        /// <summary>
+        /// The default source encoding used to parse source before producing a compilation. Only embedded in
```

Comment:
I think the word "default" here is a bit confusing. When it comes to encoding and source file reading it has a variety of interpretations: OS default, UTF8 (the implicit default) and the default provided on the command line.

My understanding is that this represents the explicit encoding provided on the command line. Perhaps use "Explicit" here.

Reply:
I'm struggling here, but only because BOM is technically the "explicit" encoding for a file. Everything else is just best guess, and so this kind of is a "default". I'll think on it more though

Reply:
It's not best guess if the user specifies `/codepage` explicitly, which is what this property should reflect. The encoding specified via `/codepage` can only be overridden by BOM and nothing else.

---
In reply to: [434211758](https://github.com/dotnet/roslyn/pull/44373#discussion_r434211758) [](ancestors = 434211758)

Reply:
I would reword this slightly: "The encoding used to parse source files that do not declare their encoding via Byte Order Mark (BOM). Specified on command line as `/codepage`."

---
In reply to: [434220080](https://github.com/dotnet/roslyn/pull/44373#discussion_r434220080) [](ancestors = 434220080,434211758)